inc/Engine/Media/Lazyload/Content/Processor/SimpleHtmlDom.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Media\Lazyload\Content\Processor;
+
+use voku\helper\HtmlDomParser;
+use voku\helper\SimpleHtmlDomBlank;
+
+class SimpleHtmlDom {
+	public function add_locations_hash_to_html( $html ) {
+		$dom = HtmlDomParser::str_get_html( $html );
+
+		$body = $dom->getElementByTagName( 'body' );
+
+		if ( $body instanceof SimpleHtmlDomBlank ) {
+			return $html;
+		}
+
+		$this->add_hash_to_element( $body, 2 );
+
+		return $dom->save();
+	}
+
+	private function add_hash_to_element( $element, $depth ) {
+		if ( $depth < 0 ) {
+			return;
+		}
+
+		$skip_tags = [
+			'DIV',
+			'MAIN',
+			'FOOTER',
+			'SECTION',
+			'ARTICLE',
+			'HEADER',
+		];
```

Comment:
Same as my previous comment, wouldn't be more interesting to put it in a class properties ?